### PR TITLE
Fixes #2182 "create new map" error on loading the map

### LINKED ISSUE #2182: Cannot create new maps
The button "Create map" opens an empty map with the message "loading..." forever.

Reproduced in DEV with Chrome and Firefox

### PR DESCRIPTION
**Pull request by MV88**
#2182 

Wanted to test the params passed to the loadMapConfig thunk.
Moved the mapViewerPage in an unconnected component for test purposes.

### CODE REVIEW

File: web/client/product/components/viewer/MapViewer.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+const React = require('react');
+const PropTypes = require('prop-types');
+const url = require('url');
+const urlQuery = url.parse(window.location.href, true).query;
```
Comment (offtherailz) [inline web/client/product/components/viewer/MapViewer.jsx]: move url parse in the body

File: web/client/product/components/viewer/__tests__/MapViewer-test.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            loadMapConfig: (cfgUrl, mapId) => {
+                expect(cfgUrl).toBe("/mapstore/rest/geostore/data/1");
+                expect(mapId).toBe(1);
+            }};
+        const component = renderMapViewerComp(mapViewerPros);
+        expect(component).toExist();
+        done();
+    });
+/*
+    it('testing creation with mapId = 0', () => {
```
Comment (offtherailz) [inline web/client/product/components/viewer/__tests__/MapViewer-test.jsx]: remove unused code

File: web/client/product/components/viewer/__tests__/MapViewer-test.jsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        const mapViewerPros = { match, location, onInit: () => {},
+            loadMapConfig: (cfgUrl, mapId) => {
+                expect(cfgUrl).toBe("/mapstore/rest/geostore/data/1");
+                expect(mapId).toBe(1);
+            }};
+        const component = renderMapViewerComp(mapViewerPros);
+        expect(component).toExist();
+        done();
+    });
+/*
```
Comment (mbarto) [inline web/client/product/components/viewer/__tests__/MapViewer-test.jsx]: If they are not running, remove them

File: web/client/product/components/viewer/MapViewer.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            let mapId = this.props.match.params.mapId === '0' ? null : this.props.match.params.mapId;
+            let config = urlQuery && urlQuery.config || null;
+            const {configUrl} = ConfigUtils.getConfigUrl({mapId, config});
+            mapId = mapId === 'new' ? null : mapId;
+            this.props.onInit();
+            this.props.loadMapConfig(configUrl, mapId);
+        }
+    }
+    render() {
+        return (<MapViewer
```
Comment (offtherailz) [inline web/client/product/components/viewer/MapViewer.jsx]: Pass this as children (or as prop) to make this class not dependent from a container

File: web/client/product/components/viewer/__tests__/MapViewer-test.jsx (hunk 3 above)
Comment (MV88) [reply web/client/product/components/viewer/__tests__/MapViewer-test.jsx]: sure

**Pull request merged**